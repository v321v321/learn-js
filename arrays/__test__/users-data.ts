export interface User {
  name: string;
  experience: number;
}

export const usersData = [
  { name: 'Zemlak', experience: 14 },
  { name: 'Upton', experience: 13 },
  { name: 'Wisoky', experience: 3 },
  { name: 'Balistreri', experience: 8 },
  { name: 'Braun', experience: 13 },
  { name: 'Walsh', experience: 9 },
  { name: 'Cummings', experience: 3 },
  { name: 'Johnston', experience: 8 },
  { name: 'Dickens', experience: 14 },
  { name: 'Treutel', experience: 6 },
  { name: 'Buckridge', experience: 12 },
  { name: 'Veum', experience: 13 },
  { name: 'Turcotte', experience: 10 },
  { name: 'Bartell', experience: 8 },
  { name: 'Gottlieb', experience: 7 },
  { name: 'Will', experience: 15 },
  { name: 'Gleason', experience: 14 },
  { name: 'Nikolaus', experience: 8 },
  { name: 'Durgan', experience: 4 },
  { name: 'Mayert', experience: 14 },
  { name: 'Flatley', experience: 8 },
  { name: 'Koelpin', experience: 3 },
  { name: 'Feest', experience: 14 },
  { name: 'Thiel', experience: 10 },
  { name: 'Collins', experience: 8 },
  { name: 'Hudson', experience: 6 },
  { name: 'Crona', experience: 7 },
  { name: 'Kihn', experience: 13 },
  { name: 'Murray', experience: 8 },
  { name: 'Rempel', experience: 2 },
  { name: 'Gleichner', experience: 6 },
  { name: 'Keebler', experience: 10 },
  { name: 'Abernathy', experience: 4 },
  { name: 'Wolf', experience: 4 },
  { name: 'Fadel', experience: 11 },
  { name: 'Simonis', experience: 15 },
  { name: 'Ullrich', experience: 7 },
  { name: 'Jones', experience: 5 },
  { name: 'Langworth', experience: 4 },
  { name: 'Lind', experience: 14 },
  { name: 'Kuphal', experience: 9 },
  { name: 'Jerde', experience: 4 },
  { name: 'Fritsch', experience: 4 },
  { name: 'Stanton', experience: 3 },
  { name: 'Turner', experience: 9 },
  { name: 'Nolan', experience: 9 },
  { name: 'Gerlach', experience: 3 },
  { name: 'Boyle', experience: 10 },
  { name: 'Pfannerstill', experience: 6 },
  { name: 'Cormier', experience: 9 },
  { name: 'Padberg', experience: 12 },
  { name: 'Purdy', experience: 13 },
  { name: 'Ritchie', experience: 4 },
  { name: 'Doyle', experience: 12 },
  { name: 'Reichert', experience: 13 },
  { name: 'Breitenberg', experience: 13 },
  { name: 'Erdman', experience: 11 },
  { name: 'Schumm', experience: 9 },
  { name: 'Pfeffer', experience: 3 },
  { name: 'VonRueden', experience: 5 },
  { name: 'Wisozk', experience: 10 },
  { name: 'Metz', experience: 8 },
  { name: 'Barton', experience: 10 },
  { name: 'Ondricka', experience: 9 },
  { name: 'Corkery', experience: 14 },
  { name: 'Okuneva', experience: 13 },
  { name: 'Brown', experience: 13 },
  { name: 'Jacobson', experience: 12 },
  { name: 'Hills', experience: 5 },
  { name: 'Lueilwitz', experience: 4 },
  { name: 'Ziemann', experience: 10 },
  { name: 'Beier', experience: 8 },
  { name: 'Lemke', experience: 5 },
  { name: 'Altenwerth', experience: 2 },
  { name: 'Keeling', experience: 7 },
  { name: 'Crist', experience: 4 },
  { name: 'Deckow', experience: 3 },
  { name: 'Pollich', experience: 6 },
  { name: 'Gusikowski', experience: 7 },
  { name: 'Stehr', experience: 5 },
  { name: 'Jast', experience: 9 },
  { name: 'Nicolas', experience: 4 },
  { name: 'Hermiston', experience: 2 },
  { name: 'Moore', experience: 12 },
  { name: 'Kautzer', experience: 2 },
  { name: 'Huel', experience: 15 },
  { name: 'Heidenreich', experience: 8 },
  { name: 'Conn', experience: 4 },
];
