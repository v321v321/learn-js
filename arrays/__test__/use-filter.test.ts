import { assert, expect, test } from 'vitest';
import {
  getUsersWithTenYeasExperienceAndNameStartS,
  getUsersWithExperienceBiggerTenYears,
  getUsersWithExperienceSmallerFiveYears,
} from '../use-filter';
import { usersData } from './users-data';

test('Get users with 10 years experience and name starts with "S"', () => {
  expect(
    getUsersWithTenYeasExperienceAndNameStartS(usersData)
  ).toMatchSnapshot();
});

test('Get users with experience bigger then 10 years', () => {
  expect(getUsersWithExperienceBiggerTenYears(usersData)).toMatchSnapshot();
});

test('Get users with experience smaller then 5 years', () => {
  expect(getUsersWithExperienceSmallerFiveYears(usersData)).toMatchSnapshot();
});
