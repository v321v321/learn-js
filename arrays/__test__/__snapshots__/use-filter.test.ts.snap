// Vitest Snapshot v1

exports[`Get users with 10 years experience and name starts with "S" 1`] = `
[
  {
    "experience": 15,
    "name": "Simonis",
  },
]
`;

exports[`Get users with bigger then 10 years experience and name starts with "S" 1`] = `
[
  {
    "experience": 15,
    "name": "Simonis",
  },
]
`;

exports[`Get users with experience bigger then 10 years 1`] = `
[
  {
    "experience": 14,
    "name": "Zemlak",
  },
  {
    "experience": 13,
    "name": "Upton",
  },
  {
    "experience": 13,
    "name": "Braun",
  },
  {
    "experience": 14,
    "name": "Dickens",
  },
  {
    "experience": 12,
    "name": "Buckridge",
  },
  {
    "experience": 13,
    "name": "Veum",
  },
  {
    "experience": 15,
    "name": "Will",
  },
  {
    "experience": 14,
    "name": "Gleason",
  },
  {
    "experience": 14,
    "name": "Mayert",
  },
  {
    "experience": 14,
    "name": "Feest",
  },
  {
    "experience": 13,
    "name": "Kihn",
  },
  {
    "experience": 11,
    "name": "Fadel",
  },
  {
    "experience": 15,
    "name": "Simonis",
  },
  {
    "experience": 14,
    "name": "Lind",
  },
  {
    "experience": 12,
    "name": "Padberg",
  },
  {
    "experience": 13,
    "name": "Purdy",
  },
  {
    "experience": 12,
    "name": "Doyle",
  },
  {
    "experience": 13,
    "name": "Reichert",
  },
  {
    "experience": 13,
    "name": "Breitenberg",
  },
  {
    "experience": 11,
    "name": "Erdman",
  },
  {
    "experience": 14,
    "name": "Corkery",
  },
  {
    "experience": 13,
    "name": "Okuneva",
  },
  {
    "experience": 13,
    "name": "Brown",
  },
  {
    "experience": 12,
    "name": "Jacobson",
  },
  {
    "experience": 12,
    "name": "Moore",
  },
  {
    "experience": 15,
    "name": "Huel",
  },
]
`;

exports[`Get users with experience smaller then 5 years 1`] = `
[
  {
    "experience": 3,
    "name": "Wisoky",
  },
  {
    "experience": 3,
    "name": "Cummings",
  },
  {
    "experience": 4,
    "name": "Durgan",
  },
  {
    "experience": 3,
    "name": "Koelpin",
  },
  {
    "experience": 2,
    "name": "Rempel",
  },
  {
    "experience": 4,
    "name": "Abernathy",
  },
  {
    "experience": 4,
    "name": "Wolf",
  },
  {
    "experience": 4,
    "name": "Langworth",
  },
  {
    "experience": 4,
    "name": "Jerde",
  },
  {
    "experience": 4,
    "name": "Fritsch",
  },
  {
    "experience": 3,
    "name": "Stanton",
  },
  {
    "experience": 3,
    "name": "Gerlach",
  },
  {
    "experience": 4,
    "name": "Ritchie",
  },
  {
    "experience": 3,
    "name": "Pfeffer",
  },
  {
    "experience": 4,
    "name": "Lueilwitz",
  },
  {
    "experience": 2,
    "name": "Altenwerth",
  },
  {
    "experience": 4,
    "name": "Crist",
  },
  {
    "experience": 3,
    "name": "Deckow",
  },
  {
    "experience": 4,
    "name": "Nicolas",
  },
  {
    "experience": 2,
    "name": "Hermiston",
  },
  {
    "experience": 2,
    "name": "Kautzer",
  },
  {
    "experience": 4,
    "name": "Conn",
  },
]
`;
